e2e-tests/src/test/validators_rotate.rs

```diff
@@ -1,78 +1,29 @@
+    wait_for_finalized_block, wait_for_full_era_completion, wait_for_session,
 };
 use primitives::CommitteeSeats;
 use sp_core::Pair;
 use substrate_api_client::{AccountId, XtStatus};
 
 use crate::{
-    accounts::{accounts_seeds_to_keys, get_sudo_key, get_validators_keys, get_validators_seeds},
+    connection::get_signed_connection,
+    validators::{
+        get_non_reserved_validators, get_non_reserved_validators_for_session,
+        get_reserved_validators,
+    },
     Config,
 };
 
+const SESSION_PERIOD: u32 = 30;
 const TEST_LENGTH: u32 = 5;
 
-fn get_reserved_validators(config: &Config) -> Vec<KeyPair> {
-    get_validators_keys(config)[0..2].to_vec()
-}
-
-fn get_non_reserved_validators(config: &Config) -> Vec<KeyPair> {
-    get_validators_keys(config)[2..].to_vec()
-}
-
-fn get_non_reserved_validators_for_session(config: &Config, session: u32) -> Vec<AccountId> {
-    // Test assumption
-    const FREE_SEATS: u32 = 2;
-
-    let mut non_reserved = vec![];
-
-    let validators_seeds = get_validators_seeds(config);
-    let non_reserved_nodes_order_from_runtime = validators_seeds[2..].to_vec();
-    let non_reserved_nodes_order_from_runtime_len = non_reserved_nodes_order_from_runtime.len();
-
-    for i in (FREE_SEATS * session)..(FREE_SEATS * (session + 1)) {
-        non_reserved.push(
-            non_reserved_nodes_order_from_runtime
-                [i as usize % non_reserved_nodes_order_from_runtime_len]
-                .clone(),
-        );
-    }
-
-    accounts_seeds_to_keys(&non_reserved)
-        .iter()
-        .map(|pair| AccountId::from(pair.public()))
-        .collect()
-}
-
-fn get_authorities_for_session<C: AnyConnection>(connection: &C, session: u32) -> Vec<AccountId> {
-    const SESSION_PERIOD: u32 = 30;
-    let first_block = SESSION_PERIOD * session;
-
-    let block = connection
-        .as_connection()
-        .get_block_hash(Some(first_block))
-        .expect("Api call should succeed")
-        .expect("Session already started so the first block should be present");
-
-    connection
-        .as_connection()
-        .get_storage_value("Session", "Validators", Some(block))
-        .expect("Api call should succeed")
-        .expect("Authorities should always be present")
-}
-
 pub fn validators_rotate(config: &Config) -> anyhow::Result<()> {
-    let node = &config.node;
-    let accounts = get_validators_keys(config);
-    let sender = accounts.first().expect("Using default accounts").to_owned();
-    let connection = SignedConnection::new(node, sender);
-
-    let sudo = get_sudo_key(config);
+    let connection = get_signed_connection(config);
+    let root_connection = config.create_root_connection();
 
-    let root_connection = RootConnection::new(node, sudo);
     let reserved_validators: Vec<_> = get_reserved_validators(config)
         .iter()
         .map(|pair| AccountId::from(pair.public()))
```

Comment:
this mapping seeds to AccountId seems a common thing, so maybe we can provide some utility in `e2e-tests/src/accounts.rs` (there are already similar methods there)

Reply:
`e2e-tests/src/test/era_validators.rs` would benefit from that as well

Reply:
Done.